albogdan/skule-vote
# Messages API Endpoint

##  **MAKE SURE TO RUN `python manage.py migrate` BEFORE REVIEWING THIS PR**

## Overview

- Resolves #106
- Removed `hideable` field from `Messages` model, and the admin site. Created respective migration
- Created serializer for `Messages`
- Created API endpoint view for the `Messages` model. Returns all active messages within the current active `ElectionSession` in a list.
- Added URL for `Messages` API endpoint to `/api/messages`


## Unit Tests Created

- Created unit tests to check that `Messages` are returned only when they are active and when they are part of an active `ElectionSession`.
- Created unit tests to test if the above conditions aren't true then no `Messages` would be returned.


## Steps to QA

- Create an `ElectionSession` that is active and one that has already past / will happen in the future.
- Create some `Messages` for each of those `ElectionSessions`.
- Go to `/api/messages` and see what gets returned. Only `Messages` that are active and that are part of an active `ElectionSession` should be returned. Test with different cases as you see fit.
